### DISCUSSION TITLE
Add Dialogue ANIM AND SFF  ability to use prefixes.

### DISCUSSION THREAD

**Discussion by Orochikyocr** [ikemen-engine/Ikemen-GO#2168] [Feature Request]
### Is your feature request related to a problem? Please describe.

You added the ability to have our own sff, snd, and air files and call whatever is contained inside them using prefixes defined in the def file.  
But inside the dialogue sctrl, this ability is locked, ANIM and SND always expect a number, not letters. It will launch an error if used.  

### Describe the solution you'd like

Imagine Im a full game developer and I have different effects and sounds contained in many files because this way is easier to edit them. I have a def file just for dialogue animations, and has the prefix DIALOGUE.
Please allow the DIALOGUE prefix to be used with anim and sound.  

text3 : "<p1><p1name=Juan Carlos><anim=self,DIALOGUE9051><map=self,isActivePlayer,1,set>You dare to stand in my way? You clearly do not know who I am.<wait=30>";

### Additional context

It could be nice for full game developers to have full control of calling animations and sounds from whatever file they feel.
I know this is not exactly useful for character creators. Yet many character creators can share a common file to have dialogue between their characters sharing a single set of files.

### Search terms

dialogue, prefix, contained files

**Comment by Orochikyocr** [comment]
To add another point to the discussion.
![image](https://github.com/user-attachments/assets/30053efa-b7a2-4b54-8c37-646ec359d8ad)
This happened while calling 9000,3 sprite from Choi files.
What if I can create my own custom dialogue anim and sff file with hundreds of portraits of the most known characters in fighting games? Then proceed to create a function that detects the character's name and use my sprite/animation for Choi instead of using the one contained in the char file, risking displaying corrupted sprites.

It could be useful for many character creators.

**Comment by Kiwi1444** [comment]
> Imagine Im a full game developer and I have different effects and sounds contained in many files because this way is easier to edit them. I have a def file just for dialogue animations, and has the prefix DIALOGUE. Please allow the DIALOGUE prefix to be used with anim and sound.

Is it don't already working like this? orochi_kyo_dialogue_main.zss says:
`The prefix to call animations from the ZSS/CNS code is DIALOGUE. For example Explod{anim: F9060;} now is Explod{anim: DIALOGUE9060;}`
If not, how do I trigger the animation using orochi_kyo dialog system?

**Reply by Orochikyocr** [reply]
You can call animations using the Dialogue anim parameter. But you can ONLY use animations inside char files, you cannot call animations from FIGHTFX or your own custom AIR file because that anim parameter does not support PREFIXES.
